### PR TITLE
Update `Graphormer` and fix its `torchscript` test failures

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Update `Graphormer` and fix its `torchscript` test failures

cc @clefourrier for reference

### CODE REVIEW

File: src/transformers/models/graphormer/modeling_graphormer.py [hunk 1]
```diff
@@ -798,9 +798,11 @@ def forward(
         attn_edge_type,
         perturb=None,
         masked_tokens=None,
-        return_dict: Optional[bool] = True,
+        return_dict: Optional[bool] = None,
```
Comment (ydshieh) [inline src/transformers/models/graphormer/modeling_graphormer.py:801]: `PretrainedConfig` has default value `True` for `return_dict`. And in `transformers`, we usually use `return_dict: Optional[bool] = None` in `forward`, and adding one line in the beginning of `forward`
```python
return_dict = return_dict if return_dict is not None else self.config.use_return_dict
```
which is
```
return self.return_dict and not self.torchscript
```
to avoid torchscript failure.

File: src/transformers/models/graphormer/modeling_graphormer.py [hunk 2]
```diff
@@ -798,9 +798,11 @@ def forward(
...
         **unused
     ):
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
```
Comment (ydshieh) [inline src/transformers/models/graphormer/modeling_graphormer.py:804]: See above comment

File: src/transformers/models/graphormer/modeling_graphormer.py [hunk 3]
```diff
@@ -819,7 +821,7 @@ def forward(
             input_nodes = torch.nn.functional.linear(input_nodes, self.graph_encoder.embed_tokens.weight)
 
         if not return_dict:
-            return (input_nodes, inner_states)
+            return tuple(x for x in [input_nodes, inner_states] if x is not None)
```
Comment (ydshieh) [inline src/transformers/models/graphormer/modeling_graphormer.py:824]: Need to pop up the None values

File: src/transformers/models/graphormer/modeling_graphormer.py [hunk 4]
```diff
@@ -871,12 +875,14 @@ def forward(
...
             spatial_pos,
             attn_edge_type,
+            return_dict=True,
         )
         outputs, hidden_states = encoder_outputs["last_hidden_state"], encoder_outputs["hidden_states"]
 
         head_outputs = self.classifier(outputs)
         logits = head_outputs[:, 0, :].contiguous()
 
+        loss = None
```
Comment (ydshieh) [inline src/transformers/models/graphormer/modeling_graphormer.py:885]: `loss` need to be defined

File: tests/test_modeling_common.py [hunk 5]
```diff
@@ -746,6 +746,19 @@ def _create_and_check_torchscript(self, config, inputs_dict):
                     traced_model = torch.jit.trace(
                         model, (input_ids, bbox, image), check_trace=False
                     )  # when traced model is checked, an error is produced due to name mangling
+                elif config.__class__.__name__ == "GraphormerConfig":  # Graphormer requires additional inputs
```
Comment (ydshieh) [inline tests/test_modeling_common.py:749]: `Graphormer` needs more inputs.

**Review by ydshieh — COMMENTED** [review]
Some comments to facilitate reviews.

File: src/transformers/models/graphormer/modeling_graphormer.py [hunk 6]
```diff
@@ -871,12 +875,14 @@ def forward(
             out_degree,
             spatial_pos,
             attn_edge_type,
+            return_dict=True,
```
Comment (ydshieh) [inline src/transformers/models/graphormer/modeling_graphormer.py:878]: Pass `True` here is fine.

File: tests/test_modeling_common.py (hunk 5 above)
Comment (sgugger) [reply tests/test_modeling_common.py:749]: Really not in favor of changing this test just for Graphormer. Please overwrite the test in the Gtaphormer test file.

**Review by sgugger — COMMENTED** [review]
Thanks for the fixes. Let's not modify the common tester for a single model though (especially as we don't know yet if it's going to be used or not).

File: tests/models/graphormer/test_modeling_graphormer.py [hunk 7]
```diff
@@ -255,6 +257,92 @@ def setUp(self):
         self.model_tester = GraphormerModelTester(self)
         self.config_tester = ConfigTester(self, config_class=GraphormerConfig, has_text_modality=False)
 
+    # overwrite from common as `Graphormer` requires more input arguments
```
Comment (ydshieh) [inline tests/models/graphormer/test_modeling_graphormer.py:260]: @sgugger As requested :-)

**Review by sgugger — APPROVED** [review]
Thanks!
